app/src/main/java/com/example/basicapplicationstructure/Test.txt

```diff
@@ -0,0 +1,184 @@
+package com.example.basicapplicationstructure.data.localDataSource
+
+import androidx.room.Dao
+import com.example.basicapplicationstructure.data.remoteDataSource.User
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface UserDao {
+
+    suspend fun getUserDataBasedOnId(id: String) : User?
+
+    suspend fun addUserData(user: User)
+}
+
+package com.example.basicapplicationstructure.data.remoteDataSource
+
+data class User(
+    val id:Int,
+    val name: String,
+    val isPremiumUser: Boolean = false
+)
+
+
+package com.example.basicapplicationstructure.data
+
+import com.example.basicapplicationstructure.data.localDataSource.UserDao
+import com.example.basicapplicationstructure.data.remoteDataSource.User
+import com.example.basicapplicationstructure.network.ApiInterface
+import com.example.basicapplicationstructure.network.NetworkResponse
+import com.example.basicapplicationstructure.network.invokeOnStatus
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+class UserRepositoryImp @Inject constructor(val apiInterface: ApiInterface, val userDao: UserDao): UserRepositoryInterface {
```

Comment:
`UserRepositoryInterface` this should have been shortlisted with `UserRepository`